### ISSUE TITLE
Refactoring and merging PSL-VMI feature

### ISSUE THREAD

**Issue by mhm90** [linqs/psl#335]
I need some changes in PSL-VMI feature and I was wondering if you could help me with that to change implementation accordingly and apply the feature in the main repo to be released in next version.
I am not using PSL-CLI but PSL-Java in my project. I have a standalone application which has its lifecycle beyond PSL run and is using PSL to infer a result from some input received as a request! Consider it as a backend service which uses PSL (run a PSL job per each request and wait for it to calculate response). I need to track why my rules resulting specific response (which is what PSL-VMI is doing) and log the `JSON` output of PSL-VMI in a searchable log container with my PSL job id like Graylog. So thats why I don't need PSL-VMI to output the debug data as a file after inference. I need it to be like a callback after inference in java code and I use that data for my logging purposes. The callback could also handles the JSON debug file generation in the CLI.

**Comment by eriq-augustine** [comment]
So it sounds like the key information that you want are the ground rules/terms associated with a specific atom/prediction.
Does that sound right?

**Comment by mhm90** [comment]
Yeah, I think I need pretty much everything for the inference debugging, but hasn't all this been implemented once in the [`PSL-Viz`](https://github.com/linqs/psl/pull/278) fork? Should I start from scratch?

**Comment by eriq-augustine** [comment]
I would take a look at PSL-Viz and see what components you may want to bring over from that.
